File: Domain/PlayableContent.cs
using System.Collections.Generic;

namespace Domain
{
    public class PlayableContent : CategoryElement
    {
        public string ImageURL { get; set; }

        public double Duration { get; set; }

        public string Author { get; set; }

        public string ContentURL { get; set; }

        public ICollection<Playlist> Playlists { get; set; }
    }
}

File: bettercalm-frontend/src/app/models/playableContent.ts
import { Injectable } from '@angular/core';
import { Adapter } from './adapter';

export class PlayableContent {
  constructor(
    public id: number,
    public name: string,
    public imageURL: string,
    public contentURL: string,
    public author: string,
    public duration: number
  ){}
}

@Injectable({
  providedIn: 'root'
})
export class PlayableContentAdapter implements Adapter<PlayableContent> {
  adapt(item: any): PlayableContent {
    return new PlayableContent(
      item.id,
      item.name,
      item.imageURL,
      item.contentURL,
      item.author,
      item.duration
    );
  }
}
